avocado/core/loader.py

```diff
@@ -512,7 +512,7 @@ def skip_non_test(exception):
             onerror = skip_non_test
 
         for dirpath, _, filenames in os.walk(url, onerror=onerror):
-            for file_name in filenames:
+            for file_name in sorted(filenames):
```

Comment:
Maybe I misunderstood the card, but if we declare alphabetical order, we should probably keep to that not just within the directory, but always. I'd expect that as a user.